spark/src/main/scala/ai/chronon/spark/JoinBase.scala

```diff
@@ -57,7 +59,13 @@ abstract class JoinBase(joinConf: api.Join,
   private val gson = new Gson()
   // Combine tableProperties set on conf with encoded Join
   protected val tableProps: Map[String, String] =
-    confTableProps ++ Map(Constants.SemanticHashKey -> gson.toJson(joinConf.semanticHash.asJava))
+    confTableProps ++ Map(
+      Constants.SemanticHashKey -> gson.toJson(joinConf.semanticHash(excludeTopic = true).asJava),
+      Constants.SemanticHashOptionsKey -> gson.toJson(
```

Comment:
@donghanz i think we will still keep `SemanticHashKey` and `SemanticHashOptionsKey` as two separate properties in hive tables.  

- `SemanticHashKey`: will store the semantic hashes (a map from string to hash string)
- `SemanticHashOptionsKey`: will store a few flags (a map from string to boolean)

I pass `excludeTopic` to `.semanticHash()`, only because `excludeTopic` will determine the logic I use to compute the hash, the flag itself is not stored in the hash. 

Lmk if that makes sense